### PR TITLE
feat: [OIP-621]: Add regex fields to verify step fields

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/components/NodeFilteringFields/NodeFilteringFields.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+interface NodeFilteringFieldsProps {
+  allowableTypes: AllowedTypes
+}
+
+export default function NodeFilteringFields({ allowableTypes }: NodeFilteringFieldsProps): JSX.Element {
+  const { getString } = useStrings()
+
+  return (
+    <Accordion>
+      <Accordion.Panel
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/components/NodeFilteringFields/NodeFilteringFields.tsx:15]:
```suggestion
      <Accordion.Panel
        id="NodeFilteringFields"
         ......
        summary={getString('projectsOrgs.optional')}
      />
```

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/__tests__/ContinousVerificationWidget.test.tsx [hunk 2]
```diff
@@ -281,4 +281,61 @@ describe('Unit tests for ContinousVerificationWidget Utils', () => {
...
+            </div>
+          </Formik>
+        </TestWrapper>
+      )
+
+      expect(screen.getByTestId(/NodeFilteringFields-panel/)).toBeInTheDocument()
+
+      await userEvent.click(screen.getByText(/projectsOrgs.optional/))
+
+      screen.debug(container, 30000)
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/__tests__/ContinousVerificationWidget.test.tsx:314]: remove debug

Comment (praneshg239) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/__tests__/ContinousVerificationWidget.test.tsx:314]: Done

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/components/NodeFilteringFields/NodeFilteringFields.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/components/NodeFilteringFields/NodeFilteringFields.tsx:15]: Updated

File: src/modules/35-connectors/strings/strings.en.yaml [hunk 3]
```diff
@@ -469,6 +469,15 @@ cdng:
...
   failOnNoAnalysis: Fail On No Analysis
+  shouldUseNodesFromCD: Should use node details from CD
+  controlNodeLabel: Control Node
+  testNodeLabel: Test Node
+  nodeFilteringTitle: Node Filtering
+  nodeFilteringDescription:
+    Node filtering helps to reduces the number of nodes analyzed in continuous verification by specifying criteria
+    for inclusion. It improves performance, reduces costs, and increases accuracy by focusing on relevant nodes.
+  controlNodePlaceholder: Enter * for default  or specify Regex [canary-1 | canary-2]
+  testNodePlaceholder: Enter * for default  or specify Regex [test-1 | test-2]
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/strings/strings.en.yaml:480]: Why these strings are added in 35-connector module?

Comment (praneshg239) [reply src/modules/35-connectors/strings/strings.en.yaml:480]: In this existing component the strings were taken from here already, so added them here. Should I need to move these to pipelines strings?

Comment (meenakshi-harness) [reply src/modules/35-connectors/strings/strings.en.yaml:480]: Yes these are not relevant to this module. Please move other strings as well to a module where these are used.

Comment (praneshg239) [reply src/modules/35-connectors/strings/strings.en.yaml]: Updated

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 4]
```diff
@@ -1905,3 +1905,13 @@ copilot:
...
+verifyStep:
+  shouldUseNodesFromCD: Should use node details from CD
+  controlNodeLabel: Control Node
+  testNodeLabel: Test Node
+  nodeFilteringTitle: Node Filtering
+  nodeFilteringDescription:
+    Node filtering helps to reduces the number of nodes analyzed in continuous verification by specifying criteria
+    for inclusion. It improves performance, reduces costs, and increases accuracy by focusing on relevant nodes.
+  controlNodePlaceholder: Enter * for default  or specify Regex [canary-1 | canary-2]
+  testNodePlaceholder: Enter * for default  or specify Regex [test-1 | test-2]
```
Comment (meenakshi-harness) [inline src/modules/70-pipeline/strings/strings.en.yaml:1917]: Is pipeline correct module for these or 85-cv as I see the usage in that module only?

Comment (praneshg239) [reply src/modules/70-pipeline/strings/strings.en.yaml:1917]: It is used in pipelines only, not `cv`

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 5]
```diff
@@ -1905,3 +1905,13 @@ copilot:
   introduction: Introducing Harness AIDA
   helpText: Harness AI Development Assistant(AIDA) can help find the root cause of the error and suggest possible solutions.
 license: License
+verifyStep:
+  shouldUseNodesFromCD: Should use node details from CD
+  controlNodeLabel: Control Node
+  testNodeLabel: Test Node
+  nodeFilteringTitle: Node Filtering
+  nodeFilteringDescription:
+    Node filtering helps to reduces the number of nodes analyzed in continuous verification by specifying criteria
```
Comment (meenakshi-harness) [inline src/modules/70-pipeline/strings/strings.en.yaml:1914]: *reduce

Comment (praneshg239) [reply src/modules/70-pipeline/strings/strings.en.yaml]: Updated

File: src/modules/70-pipeline/strings/strings.en.yaml (hunk 4 above)
Comment (praneshg239) [reply src/modules/70-pipeline/strings/strings.en.yaml:1917]: Moved the string to `cv`

**Review by meenakshi-harness — DISMISSED** [review]
Approving PL owned files

**Review by meenakshi-harness — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**